modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/detail/ReferralDetail.js

```diff
@@ -9,8 +9,216 @@
+									<Hint
+										message={Liferay.Language.get(
+											'top-referring-pages-help'
+										)}
+										title={Liferay.Language.get(
+											'top-referring-pages'
+										)}
+									/>
+								</span>
+							</span>
+						</ClayList.ItemTitle>
+					</ClayList.ItemField>
+					<ClayList.ItemField>
+						<ClayList.ItemTitle>
+							<span>{Liferay.Language.get('traffic')}</span>
+						</ClayList.ItemTitle>
+					</ClayList.ItemField>
+				</ClayList.Item>
+				{referringPages
+					.slice(0, isExpanded ? 10 : ITEMS_TO_SHOW)
+					.map(({traffic, url}) => {
+						return (
+							<ClayList.Item flex key={url}>
+								<ClayList.ItemField expand>
+									<ClayList.ItemText>
+										<ClayTooltipProvider>
+											<span
+												className="text-truncate-inline"
+												data-tooltip-align="top"
+												title={url}
+											>
+												<a
+													className="c-mr-2 text-primary text-truncate text-truncate-reverse"
+													href={url}
+													target="_blank"
+												>
+													{url}
+												</a>
+											</span>
+										</ClayTooltipProvider>
+									</ClayList.ItemText>
+								</ClayList.ItemField>
+								<ClayList.ItemField expand>
+									<span className="align-self-end">
+										{numberFormat(languageTag, traffic)}
+									</span>
+								</ClayList.ItemField>
+							</ClayList.Item>
+						);
+					})}
+			</ClayList>
+
+			<ClayButton
+				borderless
+				className="c-mb-4"
+				displayType="secondary"
+				onClick={() => setIsExpanded(!isExpanded)}
+				small
+			>
+				{isExpanded ? (
+					<span>{Liferay.Language.get('view-less')}</span>
+				) : (
+					<span>{Liferay.Language.get('view-more')}</span>
+				)}
+			</ClayButton>
+
+			<ClayList className="list-group-pages-list">
+				<ClayList.Item flex>
+					<ClayList.ItemField expand>
+						<ClayList.ItemTitle className="text-truncate-inline">
+							<span className="text-truncate">
+								{Liferay.Language.get('top-referring-domains')}
+								<span className="text-secondary">
+									<Hint
+										message={Liferay.Language.get(
+											'top-referring-domains-help'
+										)}
+										title={Liferay.Language.get(
+											'top-referring-domains'
+										)}
+									/>
+								</span>
+							</span>
+						</ClayList.ItemTitle>
+					</ClayList.ItemField>
+					<ClayList.ItemField>
+						<ClayList.ItemTitle>
+							<span>{Liferay.Language.get('traffic')}</span>
+						</ClayList.ItemTitle>
+					</ClayList.ItemField>
+				</ClayList.Item>
+				{referringDomains.map(({traffic, url}) => {
+					return (
+						<ClayList.Item flex key={url}>
+							<ClayList.ItemField expand>
+								<ClayList.ItemText>
+									<ClayTooltipProvider>
+										<span
+											className="text-truncate-inline"
+											data-tooltip-align="top"
+											title={url}
+										>
+											<a
+												className="text-primary text-truncate"
```

Comment:
I should add a margin here too. I'll update the changes tomorrow and let the CI finish.